core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/waiters/WaiterExecutorHelper.java

```diff
@@ -32,12 +34,20 @@
  */
 @SdkInternalApi
 public final class WaiterExecutorHelper<T> {
+    private static final int DEFAULT_MAX_ATTEMPTS = 3;
+    private static final BackoffStrategy DEFAULT_BACKOFF_STRATEGY = FixedDelayBackoffStrategy.create(Duration.ofSeconds(2));
```

Comment:
I guess this is kind of arbitrary, but 3 attempts with 2 second delay isn't very long to wait. I guess better to be conservative than steal the customer's thread for a long time. Maybe just maxWaitTime of 1 minute?

Reply:
Yeah, we don't want to send too many requests to the service in a short amount of time or hold the resource for too long. 

so 5-seconds delay with maxWaitTime of 1 minute?

Reply:
If we're going to make maxWaitTime similar to requestTimeout, then I think you were actually on the right track here with having requestTimeout off by default and using maxAttempts.

Maybe 5 seconds delay with max attempts of 3?

Reply:
+1 Yeah, it seems waitTimeout makes more sense since it's default to be off for generated service waiters.